### PR TITLE
PERF-5176 Add workload reproducing an OOM due to a sort-skip multi-planning storm

### PR DESCRIPTION
**Pull request by robsndr**
Jira Ticket: [PERF-5176](https://jira.mongodb.org/browse/PERF-5176)

Whats Changed:
Adds a new workload to test the server under a sort-skip multi-planning storm.

Workload Submission form:
If applicable, only required if there is a new workload being added. Form can be found [here](https://docs.google.com/forms/d/1r0kOHvFUa7rJxVmX_wp9prxYU5K155yKyZ1y3d5yhZg/edit)

### CODE REVIEW

File: src/workloads/scale/MultiPlanStormSortSkip.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+          - keys: {codes.tdsReturnId: 1, _id: -1}
+          - keys: {codes.tanId: 1, _id: -1}
+          - keys: {uniqueId: 1}
+          - keys: {codes.tdsReturnId: 1, hasDeducteeChanged: 1}
+          - keys: {codes.tanId: 1, codes.tdsReturnId: 1}
+          - keys: {codes.tanId: 1, hasPanChanged: 1}
+          - keys: {codes.tdsReturnId: 1, codes.uniqueId: 1}
+          - keys: {codes.tanId: 1, uniqueId: 1}
+
+- Name: MultiPlanStorm
```
Comment (josefahmad) [inline src/workloads/scale/MultiPlanStormSortSkip.yml:93]: How long does this phase roughly take to OOM the system? What about the whole workload, including the insertion phase?

Comment (robsndr) [reply src/workloads/scale/MultiPlanStormSortSkip.yml:93]: It takes ~10min to insert the data and build the indexes. After the InsertData phase completes, around 10s are needed to OOM the server in the MultiPlanStorm phase.

Comment (josefahmad) [reply src/workloads/scale/MultiPlanStormSortSkip.yml:93]: Sounds good, thanks!

**Review by josefahmad — APPROVED** [review]
LGTM!

**Comment by johndaniels** [comment]
@thessem FYI

**Comment by robsndr** [comment]
evergreen merge

**Comment by thessem** [comment]
The need for my team to review the dry run file is being fixed in https://github.com/mongodb/genny/pull/1186

**Pull request merged**